### ISSUE TITLE
SC2154 from strict bash mode example using trap

### ISSUE THREAD

**Issue by peterjc** [koalaman/shellcheck#2980]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154 
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
set -uo pipefail
trap 's=$?; echo "$0: Error on line "$LINENO": $BASH_COMMAND"; exit $s' ERR
IFS=$'\n\t'

echo "Example of strict bash mode from Michael Daffin, see"
echo "https://olivergondza.github.io/2019/10/01/bash-strict-mode.html"
```

#### Here's what shellcheck currently says:

```
[Line 3:](javascript:setPosition(3, 6))
trap 's=$?; echo "$0: Error on line "$LINENO": $BASH_COMMAND"; exit $s' ERR
     ^-- [SC2154](https://www.shellcheck.net/wiki/SC2154) (warning): s is referenced but not assigned.
```

#### Here's what I wanted or expected to see:

No error, my understanding is this does define ``s``. Currently https://www.shellcheck.net/wiki/SC2154 currently says:

> ShellCheck intentionally does not attempt to figure out runtime or dynamic assignments like with `source "$(date +%F).sh"` or `eval var=value`. See [SC2034](https://www.shellcheck.net/wiki/SC2034) for an extended discussion of why this is the case.

Is using `trap` equivalent, and if so it that worth adding to the documentation?

Is the recommendation as per the SC2034 documentation to use a throw-away variable ``_`` like so:

```sh
#!/bin/bash
set -uo pipefail
trap '_=$?; echo "$0: Error on line "$LINENO": $BASH_COMMAND"; exit $_' ERR
IFS=$'\n\t'

echo "Example of strict bash mode from Michael Daffin, see"
echo "https://olivergondza.github.io/2019/10/01/bash-strict-mode.html"
```

This passes shellcheck 🚀

**Comment by taiki-e** [comment]
FYI, workaround is using `trap -- '...'`.
